src/agent/onefuzz-agent/src/tasks/fuzz/libfuzzer_fuzz.rs

```diff
@@ -323,26 +323,48 @@ async fn report_fuzzer_sys_info(worker_id: usize, run_id: Uuid, fuzzer_pid: u32)
             break;
         }
 
-        time::delay_for(PROC_INFO_PERIOD).await;
+        delay_for(PROC_INFO_PERIOD).await;
     }
 
     Ok(())
 }
 
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Debug)]
 pub struct RuntimeStats {
     worker_id: usize,
     run_id: Uuid,
     count: u64,
     execs_sec: f64,
 }
 
-impl RuntimeStats {
-    pub fn report(&self) {
+#[derive(Debug, Default)]
+pub struct TotalStats {
+    worker_stats: HashMap<usize, RuntimeStats>,
+    count: u64,
+    execs_sec: f64,
+}
+
+impl TotalStats {
+    fn update(&mut self, worker_data: RuntimeStats) {
+        if let Some(current) = self.worker_stats.get(&worker_data.worker_id) {
+            // if it's the same run, only add the differences
+            if current.run_id == worker_data.run_id {
+                self.count += worker_data.count - current.count;
```

Comment:
Just to be cautious:
```suggestion
                self.count += worker_data.count.saturating_sub(current.count);
```